### PR TITLE
Add ellipse routines. Fix #146

### LINKED ISSUE #146: Graphics function for drawing arcs
The below functions should be implemented 

    gfx_Arc(<tbd>);
    gfx_FillArc(<tbd>);
    gfx_Arc_NoClip(<tbd>);
    gfx_FillArc_NoClip(<tbd>);

### PR DESCRIPTION
**Pull request by PeterTillema**

### CODE REVIEW

File: src/graphx/graphx.h [hunk 1]
```diff
@@ -756,6 +757,55 @@ void gfx_FillCircle_NoClip(uint24_t x,
...
+ * Performs faster than using gfx_FillEllipse, but can cause corruption if used outside the bounds of the screen.
+ * @param x X coordinate.
+ * @param y Y coordinate.
+ * @param a The horizontal radius of the ellipse.
+ * @param b The vertical radius of the ellipse.
+ */
+void gfx_FillEllipse_NoClip(int x,
+                            int y,
+                            int a,
+							int b);
```
Comment (adriweb) [inline src/graphx/graphx.h:773]:
```suggestion
void gfx_FillEllipse_NoClip(int x, int y, int a, int b);
```

**Review by adriweb — CHANGES_REQUESTED** [review]
nice

File: src/graphx/graphx.asm [hunk 2]
```diff
@@ -133,6 +133,13 @@ library 'GRAPHX', 11
 ; v10 functions
 ;-------------------------------------------------------------------------------
 	export gfx_CopyRectangle
+;-------------------------------------------------------------------------------
+; v11 functions
```
Comment (commandblockguy) [inline src/graphx/graphx.asm:137]: Should this be v12 or are the comments off by a version?

Comment (runer112) [reply src/graphx/graphx.asm:137]: I had a look through the history and it appears that version 10 added no functions. So yes, this should be v12. And the section above should be v11.

**Comment by mateoconlechuga** [comment]
For integer overflows, just say in the documentation what the maximum radius can be.

File: src/graphx/graphx.h [hunk 3]
```diff
@@ -756,6 +757,72 @@ void gfx_FillCircle_NoClip(uint24_t x,
...
+ * @warning
+ * When using too large radii (> 128) the upper and lower half of the ellipse might get corrupted due to
+ * integer overflow.
+ * 
+ * @param x X coordinate of the center.
+ * @param y Y coordinate of the center.
+ * @param a The horizontal radius of the ellipse.
+ * @param b The vertical radius of the ellipse.
+ */
+void gfx_FillEllipse(int x, int y, int a, int b);
```
Comment (runer112) [inline src/graphx/graphx.h:791]: Please use the following types: `int24_t x, int24_t y, uint8_t a, uint8_t b`.

The same applies to `gfx_Ellipse`.

**Review by runer112 — CHANGES_REQUESTED** [review]

File: src/graphx/graphx.h (hunk 3 above)
Comment (mateoconlechuga) [reply src/graphx/graphx.h:791]: > Please use the following types: `int24_t x, int24_t y, uint8_t a, uint8_t b`.
> 
> Also applies to `gfx_Ellipse`.

Pretty sure the radii should be int24_t?

File: src/graphx/graphx.h [hunk 4]
```diff
@@ -756,6 +757,72 @@ void gfx_FillCircle_NoClip(uint24_t x,
...
+ * @warning
+ * When using too large radii (> 128) the upper and lower half of the ellipse might get corrupted due to
+ * integer overflow.
+ * 
+ * @param x X coordinate of the center.
+ * @param y Y coordinate of the center.
+ * @param a The horizontal radius of the ellipse.
+ * @param b The vertical radius of the ellipse.
+ */
+void gfx_FillEllipse_NoClip(int x, int y, int a, int b);
```
[the remark this answers is not in the document]
Comment (mateoconlechuga) [reply src/graphx/graphx.h:775]: > Please use the following types: `uint24_t x, uint24_t y, uint8_t a, uint8_t b`.
> 
> Also applies to `gfx_Ellipse_NoClip`.

Pretty sure the radii should be `uint24_t`?

[the remark this answers is not in the document]
Comment (runer112) [reply src/graphx/graphx.h:775]: I checked the implementation beforehand. It only reads the radii as one byte values.

[the remark this answers is not in the document]
Comment (mateoconlechuga) [reply src/graphx/graphx.h:775]: Well that needs to be fixed then. I'm not even sure why he is stating there are issues "when the radii gets too large". Needs more review.

**Comment by mateoconlechuga** [comment]
Why did you close this?

File: src/graphx/graphx.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (runer112) [reply src/graphx/graphx.h:775]: Hmm... I guess that's true. It's probably a bit better that the radii be `uint24_t` then.

**Comment by mateoconlechuga** [comment]
Note for PT_: When I say that the fact that only radii of less than 128 is trash, it is just that - I am not referencing your code, I truly appreciate the pull request and think it is well done, and I am only saying that the minor issue that it doesn't support large radii is just a defect that will have to be resolved at one point because users will want it in the future. I'm not being personal about it, and I regret pointing it out, I should have handled that better. Anyway, the rest of it looks good so it can be merged and functionality added later if no one else minds.

**Comment by mateoconlechuga** [comment]
I have merged this PR in manually - I left the radii as `uint8_t` because it can be changed in the future as needed. Thanks for the PR, sorry I made such a mess of it.

**Pull request closed without merging**